### DISCUSSION TITLE
Renovate re-opens closed Chakra-UI monorepo PR with artifact errors

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#15672] [Request Help] [answered]
## How are you running Renovate?

WhiteSource Renovate hosted app on github.com

## Renovate version used

`32.64.1`

## Was this something which used to work for you, and then stopped?

It used to work, and then stopped

## Describe the bug

Renovate keeps opening a PR for the branch `renovate/major-chakra-ui-monorepo`.
The `package.json` has the same changes multiple times.
Renovate doesn't update the `package-lock.json` file in these PRs.

Pinging @Belco90, so they know I'm trying to resolve this bug. 😉 

## Relevant logs

Relevant snippet from log `#673320273`.

<details><summary>Logs</summary>

```
DEBUG: Ensuring PR(branch="renovate/major-chakra-ui-monorepo")
DEBUG: There are 0 errors and 0 warnings(branch="renovate/major-chakra-ui-monorepo")
DEBUG: getBranchPr(renovate/major-chakra-ui-monorepo)(branch="renovate/major-chakra-ui-monorepo")
DEBUG: findPr(renovate/major-chakra-ui-monorepo, undefined, open)(branch="renovate/major-chakra-ui-monorepo")
DEBUG: findPr(renovate/major-chakra-ui-monorepo, undefined, closed)(branch="renovate/major-chakra-ui-monorepo")
DEBUG: Found PR #918(branch="renovate/major-chakra-ui-monorepo")
DEBUG: Forcing PR because of artifact errors(branch="renovate/major-chakra-ui-monorepo")
DEBUG: Creating PR(branch="renovate/major-chakra-ui-monorepo")
{
  "prTitle": "Update chakra-ui monorepo (major)"
}
DEBUG: Creating PR(branch="renovate/major-chakra-ui-monorepo")
{
  "title": "Update chakra-ui monorepo (major)",
  "head": "octoclairvoyant:renovate/major-chakra-ui-monorepo",
  "base": "main",
  "draft": false
}
DEBUG: PR created(branch="renovate/major-chakra-ui-monorepo")
{
  "pr": 935,
  "draft": false
}
DEBUG: Adding labels '' to #935(branch="renovate/major-chakra-ui-monorepo")
INFO: PR created(branch="renovate/major-chakra-ui-monorepo")
{
  "pr": 935,
  "prTitle": "Update chakra-ui monorepo (major)"
}
DEBUG: addParticipants(pr=935)(branch="renovate/major-chakra-ui-monorepo")
DEBUG: Created Pull Request #935(branch="renovate/major-chakra-ui-monorepo")
WARN: artifactErrors(branch="renovate/major-chakra-ui-monorepo")
{
  "artifactErrors": [
    {
      "lockFile": "package-lock.json",
      "stderr": "npm ERR! code ERESOLVE\nnpm ERR! ERESOLVE unable to resolve dependency tree\nnpm ERR! \nnpm ERR! While resolving: octoclairvoyant-webapp@1.0.0\nnpm ERR! Found: react@17.0.2\nnpm ERR! node_modules/react\nnpm ERR!   react@\"17.0.2\" from the root project\nnpm ERR!   peer react@\">=16.8.0\" from @emotion/react@11.9.0\nnpm ERR!   node_modules/@emotion/react\nnpm ERR!     @emotion/react@\"11.9.0\" from the root project\nnpm ERR!     peer @emotion/react@\"^11.0.0\" from @chakra-ui/react@2.0.2\nnpm ERR!     node_modules/@chakra-ui/react\nnpm ERR!       @chakra-ui/react@\"2.0.2\" from the root project\nnpm ERR!     1 more (@emotion/styled)\nnpm ERR!   3 more (@emotion/styled, framer-motion, react-dom)\nnpm ERR! \nnpm ERR! Could not resolve dependency:\nnpm ERR! peer react@\">=18\" from @chakra-ui/react@2.0.2\nnpm ERR! node_modules/@chakra-ui/react\nnpm ERR!   @chakra-ui/react@\"2.0.2\" from the root project\nnpm ERR! \nnpm ERR! Fix the upstream dependency conflict, or retry\nnpm ERR! this command with --force, or --legacy-peer-deps\nnpm ERR! to accept an incorrect (and potentially broken) dependency resolution.\nnpm ERR! \nnpm ERR! See /tmp/renovate-cache/others/npm/eresolve-report.txt for a full report.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /tmp/renovate-cache/others/npm/_logs/2022-05-21T01_47_46_515Z-debug-0.log\n"
    }
  ]
}
DEBUG: Getting comments for #935(branch="renovate/major-chakra-ui-monorepo")
DEBUG: Found 0 comments(branch="renovate/major-chakra-ui-monorepo")
DEBUG: Ensuring comment "⚠ Artifact update problem" in #935(branch="renovate/major-chakra-ui-monorepo")
INFO: Comment added(branch="renovate/major-chakra-ui-monorepo")
{
  "issueNo": 935,
  "topic": "⚠ Artifact update problem"
}
```

</details>

## Have you created a minimal reproduction repository?

https://github.com/renovate-reproductions/15672

## Public repo links

- https://github.com/octoclairvoyant/octoclairvoyant-webapp/pull/935 <-- This is the PR with the log info from above
- https://github.com/octoclairvoyant/octoclairvoyant-webapp/pull/934
- https://github.com/octoclairvoyant/octoclairvoyant-webapp/pull/933
- https://github.com/octoclairvoyant/octoclairvoyant-webapp/pull/931

**Comment by rarkins** [comment, marked as answer]
The PR is immortal so reopening is expected. However if the PR had "to v2" in its title then it wouldn't be immortal. Please create a minimal reproduction and we should address the PR title as a bug assuming it wasn't caused by user config

**Reply by HonkingGoose** [reply]
## PR immortal :ghost:

> The PR is immortal so reopening is expected.

I was just about to comment something about the :ghost: immortal status of the PR. I didn't notice this in my haste to get a bug report out. Sorry about that!

## About `to v2` for monorepos

> However if the PR had "to v2" in its title then it wouldn't be immortal. Please create a minimal reproduction and we should address the PR title as a bug assuming it wasn't caused by user config

Should monorepo PRs get a `to v2` in their title? The change in the `package.json` looks like this:

```diff
 "dependencies": {
-    "@chakra-ui/react": "1.8.8",
+    "@chakra-ui/react": "2.0.2",

... snip ...

"devDependencies": {
-   "@chakra-ui/cli": "1.9.0",
+   "@chakra-ui/cli": "2.0.0",
```

The update bundles a dependency and a development dependency in one PR.

## Minimal reproduction request

> Please create a minimal reproduction and we should address the PR title as a bug assuming it wasn't caused by user config

Working on it.

## Noise reduction?

In the meantime, what's the best way to keep the noise down? Just keep the PR open, and "ignore it"?

**Reply by rarkins** [reply]
Set major updates to require dashboard approval. Then close the PR and delete the branch

**Reply by HonkingGoose** [reply]
@rarkins Here's the minimal reproduction, I've put it in our `renovate-reproductions` organization.

https://github.com/renovate-reproductions/15672

**Reply by rarkins** [reply]
I thought I'd seen this before.. the fix is in #15586

**Reply by HonkingGoose** [reply]
The problem is solved by #15586 and is live on the GitHub hosted app.
